src/components/main/Main.tsx

```diff
@@ -18,35 +18,37 @@ const Main: FC = () => {
   const [currentWeather, setCurrentWeather] = useState<CurrentWeatherType>(defaultCurrentWeather);
   const [hourlyWeather, setHourlyWeather] = useState<HourlyWeatherType[]>([]);
   const [dailyWeather, setDailyWeather] = useState<DailyWeatherType[]>([]);
+  const [selectDays, setSelectDays] = useState<string>('');
 
   async function fetchData(lon: number, lat: number) {
     Promise.all([
       getCity(lon, lat),
       getWeather<{ current: CurrentWeatherType }>('/current/', lon, lat),
-      getWeather<{ forecast: HourlyWeatherType[] }>('/forecast/hourly/', lon, lat),
-      getWeather<{ forecast: DailyWeatherType[] }>(
-        '/forecast/daily/',
-        lon,
-        lat,
-        '&periods=10&dataset=full'
-      )
+      getWeather<{ forecast: HourlyWeatherType[] }>('/forecast/hourly/', lon, lat)
     ]).then(res => {
       setLocation(res[0]);
       setCurrentWeather(res[1].current);
       setHourlyWeather(res[2].forecast);
-      setDailyWeather(res[3].forecast);
     });
   }
 
   useEffect(() => {
     if (position.longitude) {
-      const lon: number = position.longitude;
-      const lat: number = position.latitude;
-
-      fetchData(lon, lat);
+      fetchData(position.longitude, position.latitude);
     }
   }, [position.longitude]);
 
+  useEffect(() => {
+    if (selectDays) {
+      getWeather<{ forecast: DailyWeatherType[] }>(
+        '/forecast/daily/',
+        position.longitude,
+        position.latitude,
+        `&periods=${selectDays}&dataset=full`
```

Comment:
you can send this params to `httpService` function as object and there transform to query string